odoo/http.py

```diff
@@ -727,7 +727,9 @@ def route_wrapper(self, *args, **params):
             params_ko = set(params) - set(params_ok)
             if params_ko:
                 _logger.warning("%s called ignoring args %s", fname, params_ko)
-
+            if routing.get('captcha'):
+                if request.httprequest.method == 'POST' and not request.env['ir.http']._verify_request_recaptcha_token(routing['captcha']):
+                    raise UserError(odoo._("Suspicious activity detected by Google reCaptcha."))
```

Comment:
This is the wrong place to verify the captcha. Please move this code inside `ir.http`'s `_pre_dispatch`.